scylla/src/transport/session_test.rs

```diff
@@ -1059,3 +1059,35 @@ async fn test_timestamp() {
 
     assert_eq!(results, expected_results);
 }
+
+#[tokio::test]
+async fn test_client_timeout() {
+    use std::time::Duration;
+    let uri = std::env::var("SCYLLA_URI").unwrap_or_else(|_| "127.0.0.1:9042".to_string());
+    // Spontaneous errors are expected when running with a client timeout set to 0 seconds.
+    // If they happen, the test case is assumed to be correct
```

Comment:
I feel that it would be nice for client-side timeouts to be configurable per-query, not per-session. Latency requirements may be different for different queries - in particular, setting the timeout too low may cause some of the internal queries not to work correctly, e.g. topology refresh. Also, if we did that, this test could be written differently so that it only expects failure from the query with explicitly set client-side timeout.

We could have a separate internal timeout for queries (maybe the connection timeout could be used?)

Reply:
True, I'll add a way to define a client timeout per query